port/osx/omrosdump.c

```diff
@@ -465,7 +467,8 @@ omrdump_create(struct OMRPortLibrary *portLibrary, char *filename, char *dumpTyp
 		}
 	} /* 0 == child_pid */
 
-	return kr;
+	/* check the core file has actually been created */
+	return access(filename, F_OK);
```

Comment:
We don't want to ignore the value of `kr` here, and the return type of `access()` is incompatible with the return type of this function (actually, so is `kern_return_t`); suggest:
```
    return ((KERN_SUCCESS == kr) && (0 == access(filename, F_OK))) ? 0 : 1;
```
Other `return` statements above should either set `kr` appropriately and `goto` here, or properly map `kern_return_t` to `uintptr_t`.

Reply:
Changed it to what you suggested. For the child process though, an error will just cause it to not write the core file and thus fail the `access()` check at the end, since the value of `kr` is harder to propagate to the parent.